wordpress-mobile/WordPress-Android
# Issue/10643 material plugins

Fixes #10643

This PR moves Plugin Browser and details screens to Material Theme and adds Dark Theme support. There are some code changes for cosmetic reasons.

While refactoring is not in the scope of this PR, since I touched those files, I cleaned up/optimized some of the redundant code as suggested by AS, so do not be alarmed by the size of the PR. I suggest turning on the "Hide whitespace changes" option during the review.

There is a huge amount of PRs in this feature that covers all the screens within the app, so I'm requesting reviews from a wide variety of people.

![comparison_plugins](https://user-images.githubusercontent.com/728822/71138197-8e69f400-21bf-11ea-8f1a-70d3a43c7852.png)

To test:
- Navigate to Plugin Browser and Details screen and confirm it's visual and functional integrity in both Light and Dark themes. The theme can be changed from App Settings.

PR submission checklist:
